File: IAIso-v5.0/core/iaiso-php/src/Conformance/VectorResult.php
<?php

declare(strict_types=1);

namespace IAIso\Conformance;

/** Result of running one vector. */
final class VectorResult
{
    public function __construct(
        public readonly string $section,
        public readonly string $name,
        public readonly bool $passed,
        public readonly string $message = '',
    ) {
    }

    public static function pass(string $section, string $name): self
    {
        return new self($section, $name, true);
    }

    public static function fail(string $section, string $name, string $message): self
    {
        return new self($section, $name, false, $message);
    }
}

File: IAIso-v5.0/core/iaiso-java/iaiso-conformance/src/main/java/io/iaiso/conformance/VectorResult.java
package io.iaiso.conformance;

import java.util.ArrayList;
import java.util.List;

/** Result of running one vector. */
public final class VectorResult {
    public final String section;
    public final String name;
    public final boolean passed;
    public final String message;

    public VectorResult(String section, String name, boolean passed, String message) {
        this.section = section;
        this.name = name;
        this.passed = passed;
        this.message = message;
    }

    public static VectorResult pass(String section, String name) {
        return new VectorResult(section, name, true, "");
    }

    public static VectorResult fail(String section, String name, String message) {
        return new VectorResult(section, name, false, message);
    }
}
